invesalius/data/imagedata_utils.py

```diff
@@ -296,15 +299,17 @@ def create_dicom_thumbnails(image, window=None, level=None):
         else:
             thumb_image = zoom(np_image, (0.25, 0.25, 1))
         imageio.imsave(thumbnail_path, thumb_image)
+        os.close(fd)
         return thumbnail_path
 
 
 def array2memmap(arr, filename=None):
     if filename is None:
-        filename = tempfile.mktemp(prefix="inv3_", suffix=".dat")
+        fd, filename = tempfile.mkstemp(prefix="inv3_", suffix=".dat")
     matrix = numpy.memmap(filename, mode="w+", dtype=arr.dtype, shape=arr.shape)
     matrix[:] = arr[:]
     matrix.flush()
+    os.close(fd)
```

Comment:
you need to check if `fd` exists before `os.close`. Also, you can close the file because the file is being used by `numpy.memmap`.

Reply:
Thanks. Changed it.